compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8423,4 +8425,90 @@ private STNode parseParenthesisedTypeDesc() {
         STNode closeParen = parseCloseParenthesis();
         return STNodeFactory.createParenthesisedTypeDescriptorNode(openParen, typedesc, closeParen);
     }
+
+    /**
+     * Parse binding-patterns.
+     *
+     * binding-pattern :=
+     *   capture-binding-pattern
+     *    | wildcard-binding-pattern
+     *    | list-binding-pattern
+     *    | mapping-binding-pattern
+     *    | functional-binding-pattern
+     * capture-binding-pattern := variable-name
+     * variable-name := identifier
+     * wildcard-binding-pattern := _
+     * list-binding-pattern := [ list-member-binding-patterns ]
+     * list-member-binding-patterns :=
+     *    binding-pattern (, binding-pattern)* [, rest-binding-pattern]
+     *    | [ rest-binding-pattern ]
+     * mapping-binding-pattern := { field-binding-patterns }
+     * field-binding-patterns :=
+     *    field-binding-pattern (, field-binding-pattern)* [, rest-binding-pattern]
+     *    | [ rest-binding-pattern ] 
+     * field-binding-pattern :=
+     *    field-name : binding-pattern
+     *    | variable-name
+     * rest-binding-pattern := ... variable-name
+     * functional-binding-pattern := functionally-constructible-type-reference ( arg-list-binding-pattern )
+     * arg-list-binding-pattern :=
+     *    positional-arg-binding-patterns [, other-arg-binding-patterns]
+     *    | other-arg-binding-patterns
+     * positional-arg-binding-patterns := positional-arg-binding-pattern (, positional-arg-binding-pattern)*
+     * positional-arg-binding-pattern := binding-pattern
+     * other-arg-binding-patterns :=
+     *    named-arg-binding-patterns [, rest-binding-pattern]
+     *    | [rest-binding-pattern]
+     * named-arg-binding-patterns := named-arg-binding-pattern (, named-arg-binding-pattern)*
+     * named-arg-binding-pattern := arg-name = binding-pattern
+     *
+     * @return binding-pattern node
+     */
+    private STNode parseBindingPattern() {
+        STToken token = peek();
+
+        switch (token.kind) {
+            case IDENTIFIER_TOKEN:
+                STNode captureBindingPattern = parseCaptureBindingPattern();
+                return STNodeFactory.createBindingPatternNode(captureBindingPattern);
```

Comment:
Why do we need a wrapper?

Reply:
fair point, thats how I have done it in the other PR, I ll change it here as well.